ledger-core/v2/application/testwalletapi/handlers.go

```diff
@@ -11,40 +11,80 @@ import (
 	"fmt"
 	"net/http"
 
+	"github.com/insolar/assured-ledger/ledger-core/v2/application/builtin/proxy/testwallet"
+	"github.com/insolar/assured-ledger/ledger-core/v2/application/testwalletapi/statemachine"
+	"github.com/insolar/assured-ledger/ledger-core/v2/insolar"
 	"github.com/insolar/assured-ledger/ledger-core/v2/insolar/gen"
+	"github.com/insolar/assured-ledger/ledger-core/v2/insolar/payload"
 	"github.com/insolar/assured-ledger/ledger-core/v2/insolar/utils"
 	"github.com/insolar/assured-ledger/ledger-core/v2/instrumentation/inslogger"
+	"github.com/insolar/assured-ledger/ledger-core/v2/log"
+	"github.com/insolar/assured-ledger/ledger-core/v2/logicrunner/builtin/foundation"
+	"github.com/insolar/assured-ledger/ledger-core/v2/vanilla/throw"
 )
 
-func mustConvertMapToJSON(data map[string]interface{}) []byte {
-	jsonString, err := json.MarshalIndent(data, "", "    ")
-	if err != nil {
-		panic(err)
-	}
+type logIncomingRequest struct {
+	*log.Msg `txt:"Incoming request"`
 
-	return jsonString
+	URL string
 }
 
 const (
-	traceIDField = "traceID"
-	errorField   = "error"
+	referenceField = "reference"
+	traceIDField   = "traceID"
+	errorField     = "error"
 )
 
-func create(w http.ResponseWriter, req *http.Request) {
-	traceID := utils.RandTraceID()
-	_, inslog := inslogger.WithTraceField(context.Background(), traceID)
-	inslog.Info("Incoming request: ", req.URL)
+func (s *TestWalletServer) Create(w http.ResponseWriter, req *http.Request) {
+	var (
+		ctx     = context.Background()
+		traceID = utils.RandTraceID()
+		logger  = inslogger.FromContext(ctx)
+	)
+
+	logger.Infom(logIncomingRequest{URL: req.URL.String()})
 
 	result := map[string]interface{}{
-		"reference":  gen.Reference().String(),
-		traceIDField: traceID,
-		errorField:   nil,
+		referenceField: nil,
```

Comment:
If we going to benchmark VE perfomance using map[string] approach, it will afffect perfomance. It's better to go for struct Response and probably use something like https://github.com/json-iterator/go for serializartion

Reply:
done